utils/utils.go

```diff
@@ -146,3 +147,47 @@ func CheckThrottling(errCode int, err error) *resource.RetryError {
 	}
 	return resource.NonRetryableError(err)
 }
+
+func InterfaceSliceToStringSlicePtr(slice []interface{}) *[]string {
+	result := InterfaceSliceToStringSlice(slice)
+	return &result
+}
+
+func SetToStringSlice(set *schema.Set) []string {
+	return InterfaceSliceToStringSlice(set.List())
+}
+
+func SetToStringSlicePtr(set *schema.Set) *[]string {
+	return InterfaceSliceToStringSlicePtr(set.List())
+}
+
+func InterfaceSliceToStringSlice(slice []interface{}) []string {
+	result := make([]string, 0, len(slice))
+	for _, v := range slice {
+		val, ok := v.(string)
+		if ok && val != "" {
+			result = append(result, v.(string))
+		}
+	}
+	return result
+}
+
+func InterfaceSliceToStringList(slice []interface{}) *[]string {
+	res := InterfaceSliceToStringSlice(slice)
+	return &res
+}
+
+func StringSlicePtrToInterfaceSlice(list *[]string) []interface{} {
+	if list == nil {
+		return make([]interface{}, 0)
+	}
+	vs := make([]interface{}, 0, len(*list))
+	for _, v := range *list {
+		vs = append(vs, v)
+	}
+	return vs
+}
+
+func I32toa(i int32) string {
```

Comment:
do you really need a int32 parameter ?
can't you just take an int ?
Also maybe IntToStr would be more explicit.

Reply:
Int32ToStr ?